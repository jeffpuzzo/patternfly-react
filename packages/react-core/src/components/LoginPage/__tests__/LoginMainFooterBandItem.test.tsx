import * as React from 'react';
import { render, screen } from '@testing-library/react';
import { LoginMainFooterBandItem } from '../LoginMainFooterBandItem';

describe('LoginMainFooterBandItem', () => {
  test('renders with PatternFly Core styles', () => {
    render(<LoginMainFooterBandItem data-testid="test-id" />);
    expect(screen.getByTestId('test-id').outerHTML).toMatchSnapshot();
  });

  test('className is added to the root element', () => {
    render(<LoginMainFooterBandItem className="extra-class">test</LoginMainFooterBandItem>);
    expect(screen.getByText('test').outerHTML).toMatchSnapshot();
  });

  test('with custom node', () => {
    const CustomNode = () => <div>My custom node</div>;

    render(
      <LoginMainFooterBandItem>
        <CustomNode />
      </LoginMainFooterBandItem>
    );
    expect(screen.getByText('My custom node').outerHTML).toMatchSnapshot();
  });
});
