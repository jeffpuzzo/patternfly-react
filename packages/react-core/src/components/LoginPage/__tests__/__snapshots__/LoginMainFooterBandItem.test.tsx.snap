// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginMainFooterBandItem className is added to the root element 1`] = `"<p class=\\"pf-c-login__main-footer-band-item extra-class\\">test</p>"`;

exports[`LoginMainFooterBandItem renders with PatternFly Core styles 1`] = `"<p class=\\"pf-c-login__main-footer-band-item\\" data-testid=\\"test-id\\"></p>"`;

exports[`LoginMainFooterBandItem with custom node 1`] = `"<div>My custom node</div>"`;
